symengine/sympy_compat.py

```diff
@@ -64,158 +64,188 @@ def __new__(cls, name):
         return symengine.UndefFunction(name)
 
 
-class log(Function):
+from types import ModuleType
+
+functions = ModuleType(__name__ + ".functions")
+import sys
+sys.modules[functions.__name__] = functions
+
+functions.sqrt = sqrt
+functions.exp = exp
+
+
+class _FunctionRegistrarMeta(BasicMeta):
+
+    def __new__(mcls, name, bases, dict):
+        cls = BasicMeta.__new__(mcls, name, bases, dict)
+        if not name.startswith("_"):
+            setattr(functions, name, cls)
+        return cls
```

Comment:
I don' think I understand what you mean, could you please explain a bit more? This code is not trying to add an attribute to the expression `(sin(x) + 1) - 1)`. It's operating on the class `sympy_compat.sin`.